File: NeetCode/Python/BinaryTrees/Easy/104.MaximumDepthBinaryTree.py
# Given the root of a binary tree, return its maximum depth.

class TreeNode(object):
    def __init__(self, val=0, left=None, right=None):
        self.val = val
        self.left = left
        self.right = right

def maxDepth(root):
    if not root:
        return 0

    return 1 + max(maxDepth(root.left), maxDepth(root.right))

File: swift/Neetcode/Trees/Easy/104.MaximumDepthBinaryTree.swift
// Given the root of a binary tree, return its maximum depth.

// Time: O(n), Space: O(h)
func maxDepth(_ root: TreeNode?) -> Int {
    guard let root = root else { return 0 }

    return 1 + max(maxDepth(root.left), maxDepth(root.right))
}
